pkg/dependency/parser/golang/binary/parse.go

```diff
@@ -43,10 +44,21 @@ func (p *Parser) Parse(r xio.ReadSeekerAt) ([]types.Library, []types.Dependency,
 	}
 
 	libs := make([]types.Library, 0, len(info.Deps)+1)
-	libs = append(libs, types.Library{
-		Name:    "stdlib",
-		Version: strings.TrimPrefix(info.GoVersion, "go"),
-	})
+	libs = append(libs, []types.Library{
+		{
+			// Add the Go version used to build this binary.
+			Name:    "stdlib",
+			Version: strings.TrimPrefix(info.GoVersion, "go"),
+		},
+		{
+			// Add main module
+			Name: info.Main.Path,
+			// Only binaries installed with `go install` contain semver version of the main module.
+			// Other binaries use the `(devel)` version.
+			// See https://github.com/aquasecurity/trivy/issues/1837#issuecomment-1832523477.
+			Version: info.Main.Version,
```

Comment:
I want to use an empty string rather than `(devel)`. What do you think?

Reply:
I thought about it.
I am concerned about user questions:
`(devel)` indicates that we have determined version. It's not correct version, but it's version that the Go binary contains.
If we use empty version - some users may say: this is bug, because some dependencies don't have versions (for example #6456 - the problem here is different, but even in such situations questions arise).

But we can write about this in the documentation.

___

So I checked a bit:
looks like it work good. e.g. purl simply doesn't contain version.
CycloneDX and SPDX work well:
```json
{
      "bom-ref": "pkg:golang/github.com/aquasecurity/trivy",
      "type": "library",
      "name": "github.com/aquasecurity/trivy",
      "purl": "pkg:golang/github.com/aquasecurity/trivy",
      "properties": [
        {
          "name": "aquasecurity:trivy:PkgType",
          "value": "gobinary"
        }
      ]
    },
```
```json
"name": "github.com/aquasecurity/trivy",
      "SPDXID": "SPDXRef-Package-5290f6d6347a1886",
      "supplier": "NOASSERTION",
      "downloadLocation": "NONE",
      "filesAnalyzed": false,
      "sourceInfo": "package found in: trivy",
      "licenseConcluded": "NONE",
      "licenseDeclared": "NONE",
      "externalRefs": [
        {
          "referenceCategory": "PACKAGE-MANAGER",
          "referenceType": "purl",
          "referenceLocator": "pkg:golang/github.com/aquasecurity/trivy"
        }
      ],
      "attributionTexts": [
        "PkgType: gobinary"
      ],
      "primaryPackagePurpose": "LIBRARY"
```

So I like your solution. We will get more correct purl. The `(devel)` version will not force users to look at the docs to understand what it is.

But then we also need to overwrite `(devel)` for dependencies.